### DISCUSSION TITLE
New subsystem logging level: hash payload

### DISCUSSION THREAD

**Discussion by VirgoInno** [nordic-institute/X-Road#2439] [Ideas]
An enhancement request from SMIT:
It would be nice to have an additional logging level which logs only the hash of a payload instead of the payload itself.

The feature would be especially useful for large payloads like biometrics and the ones that contain GDPR special category data (which is prohibited to store to logs in plaintext without encryption).
Preferred hash algorithm is BLAKE2 as it is the fastest in software implementation and does not contain any serious vulnerabilities.
The feature can be implemented on X-Road sub-system level.

**Comment by raits** [comment]
Thank you for the change request. We have included it for discussion in the next X-Road Technical Committee meeting scheduled for the 25th of November.

I will update this discussion with the results.

**Reply by VirgoInno** [reply]
Any news?

**Reply by raits** [reply]
Hey, sorry for the lack of update here. The Technical Committee decided that this is a subject that needs to be looked into further. One of the main questions was if it would bring any bigger benefit to having the hash for the body in addition to the one we already have for the whole message.

We will conduct a study to look into this option further and consider the pros and cons, after that the topic will be raised at the Technical Committee again.
